### ISSUE TITLE
Task contexts (OTPL) are not set correctly

### ISSUE THREAD

**Issue by arthurhjorth** [NetLogo/NetLogo#755]
OTPL of tasks seems to be set depending on where the task is compiled, and not actually set on the contexts that ought to allow the task to be run. For instance, if I put this code in a procedure and run it:

```
to test
  let atask task [xcor] show ls:to-otpl atask
  set atask [task [xcor]] of one-of turtles show ls:to-otpl atask   
  set atask task [xcor] show ls:to-otpl atask
end
```

The result is this:

```
observer> test
observer: "OTPL"
observer: "OTPL"
observer: "OTPL"
```

But if I run the same code from the Command Center, 

```
observer>   let atask task [xcor] show ls:to-otpl atask 
set atask [task [xcor]] of one-of turtles show ls:to-otpl atask
set atask task [xcor] show ls:to-otpl atask
```

I get this:

```
observer: "O---"
observer: "O---"
observer: "O---"
```

(the java code for ls:to-otpl can be found here: https://gist.github.com/arthurhjorth/cd51ca93f37278603754)

**Comment by mrerrormessage** [comment]
I believe this is fixed in 5.3, can you confirm @arthurhjorth ?

**Comment by mrerrormessage** [comment]
Fixed in 5.3.1
